rdt/transformers/numerical.py

```diff
@@ -487,3 +490,160 @@ def _reverse_transform(self, data):
             data = self._univariate.ppf(scipy.stats.norm.cdf(data))
 
         return super()._reverse_transform(data)
+
+
+class BayesGMMTransformer(NumericalTransformer):
+    """Bayesian GMM transformer."""
+
+    STD_MULTIPLIER = 4
+    DETERMINISTIC_TRANSFORM = False
+    DETERMINISTIC_REVERSE = True
+    COMPOSITION_IS_IDENTITY = False
+
+    def __init__(self, dtype=None, nan='mean', null_column=None, max_clusters=10,
+                 weight_threshold=0.005):
+        super().__init__(dtype=dtype, nan=nan, null_column=null_column)
+        self._max_clusters = max_clusters
+        self._weight_threshold = weight_threshold
+        self._number_of_modes = None
+        self._bgm_transformer = None
+        self._valid_component_indicator = None
+
+    def get_output_types(self):
+        """Return the output types supported by the transformer.
+
+        Returns:
+            dict:
+                Mapping from the transformed column names to supported data types.
+        """
+        output_types = {
+            'continuous': 'float',
+            'discrete': 'categorical'
+        }
+        if self.null_transformer and self.null_transformer.creates_null_column():
+            output_types['is_null'] = 'float'
+
+        return self._add_prefix(output_types)
+
+    def _fit(self, data):
+        """Fit the transformer to the data.
+
+        Args:
+            data (pandas.Series):
+                Data to fit to.
+        """
+        self._bgm_transformer = BayesianGaussianMixture(
+            n_components=self._max_clusters,
+            weight_concentration_prior_type='dirichlet_process',
+            weight_concentration_prior=0.001,
+            n_init=1
+        )
+
+        super()._fit(data)
+        data = super()._transform(data)
+        if data.ndim > 1:
+            data = data[:, 0]
+
+        self._bgm_transformer.fit(data.reshape(-1, 1))
+        self._valid_component_indicator = self._bgm_transformer.weights_ > self._weight_threshold
```

Comment:
what does `_valid_component_indicator` actually mean/ what is it? If possible, could we rename it to something more straightforward?

Reply:
Each "component" is basically each gaussian that the model believes constitutes the data. Each component is assigned a weight, `_bgm_transformer.weights_`, which tells how important it is in generating the data. We eliminate the components which have a very low weight (smaller than some `_weight_threshold`). So `_valid_component_indicator` is an indicator of only the valid components, i.e. which gaussians are relevant.

I'm not sure what else we could call it, since "component" is the usual term used when discussing mixture models.

Reply:
I think that `component` is the correct term, but in other places (like the line below) we talk about `modes`, so maybe we want to uniformize this.

On the other hand I agree with @amontanez24 on this not being intuitive based on the attribute name alone. Maybe we should add an `Attributes` section on the `class` docstring where these things are explained?

Reply:
I agree, I was actually in the middle of adding the descriptions to the docs 👍